src/v2i-hub/CARMACloudPlugin/src/CARMACloudPlugin.cpp

```diff
@@ -209,7 +209,14 @@ void CARMACloudPlugin::CARMAResponseHandler(QHttpEngine::Socket *socket)
 	QString st; 
 	while(socket->bytesAvailable()>0)
 	{	
-		st.append(socket->readAll());
+		auto readBytes = socket->readAll();
+		if (socket->headers().keys().contains(CONTENT_ENCODING_KEY) && std::string(socket->headers().constFind(CONTENT_ENCODING_KEY).value().data()) == CONTENT_ENCODING_VALUE)

```

Comment:
Oh wait so you check for the gzip header in the request. If it is present you uncompress the data from the request, if not you use the old read functionality?

Reply:
yes